zephyr/lc3_config.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 Bose Corporation All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIB_LIBLC3CODEC_ZEPHYR_LC3_CONFIG_H_
+#define LIB_LIBLC3CODEC_ZEPHYR_LC3_CONFIG_H_
+
+/* This file is a c-wrapper for the c++ LC3 lib. Please refer to the C++ API documentation
+ * for a full description of the API.
+ * General usage:
+ * 1) Call _create to allocate memory for the c++ object using the c++-lib allocation mechanism.
+ * 2) Use the wrapper functions as needed passing the allocated object as first parameter.
+ * 3) After usage call _destroy to free up the memory.
+ *    Invalid usage of the object reference after a call to _destroy cannot be detected and will
+ *    most likely result in a crash. */
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+#define LC3_EXTERN_C extern "C"
+#else
+#define LC3_EXTERN_C
+#endif
+
+typedef void* Lc3Config_t;
+
+typedef enum
+{
+    lc3config_duration_10ms,
+    lc3config_duration_7p5ms
+} lc3config_frame_duration_t;
+
+
+/**
+ * For detailed description please see underlying LC3 codec implementation.
+ * @param Fs_ Sampling frequency Hz
+ * @param N_ms_ Frame duration
+ * @param Nc_ Number of channels
+ * @return the Config instance
+ */
+LC3_EXTERN_C Lc3Config_t Lc3Config_create(uint16_t Fs_, lc3config_frame_duration_t N_ms_,
```

Comment:
Why the camel-casing?  I have little experience in this, but I would expect the "c" side to use lower case identifiers with underscores.

Reply:
The idea was to keep the cpp class and functions names - adding only the _create and _destroy. This is to keep a 1:1 mapping when looking at the documentation of the CPP-classes.
So camel-case is used when referring to existing functions, classes and their members/arguments, whereas _lower-case is used when adding new functions like create and destroy.
Is that acceptable? - else I do not mind changing it :-)

Reply:
I do not have enough experience in Zephyr to say whether this is acceptable or not - I can live with it.
Let us see if e.g. @jhedberg or @carlescufi has something to say on this.

Reply:
I have not heard any input from @jhedberg nor @carlescufi, so I'll leave it as is and upload an update with your comments addressed. Just completed validation using different sample rates (up to 48kHz) and bitrates (up to 160kbps) using real audio content over a BT controller link - works just fine.